app/src/main/java/com/example/weather/DetailsFragment.kt

```diff
@@ -41,9 +46,26 @@ class DetailsFragment : Fragment() {
         return binding.root
     }
 
+    @RequiresApi(Build.VERSION_CODES.O)
     override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
         super.onViewCreated(view, savedInstanceState)
-        binding.detailsTextView.text = cityId
+        viewModel.cityData.observe(viewLifecycleOwner) {
+            binding.name.text = it?.name.toString()
```

Comment:
let's move all this binding setting logic to its own method for better readability instead of having it all inside the observe {} block